deCODE/deCODE.py

```diff
@@ -0,0 +1,132 @@
+                            (mt2['sample_qc'].dp_stats.mean >= 20)         
+        mt3 = mt2.filter_cols(filter_conditions, keep=True)
+ 
+        # Exclude related samples
+        mt4 = mt3.filter_cols(mt3['meta'].related, keep=False)
+        
+        # Skip ancestry check
+
+        # Section 3 - Variant-level QC
+        
+        mt4 = hl.variant_qc(mt4) 
+
+        # Apply AS_VQSR filters
+        # Restricted to bi-alleleic variants
+        # Exclude variants with inbreeding Coefficient < -0.3
+        filter_conditions = (hl.is_missing(mt4['allele_type'])) | \
+                            ((hl.is_defined(mt4['allele_type'])) & (mt4['allele_type']=='snv') & \
+                             (mt4['AS_VQSLOD'] < mt4['filtering_model'].snv_cutoff.min_score)) | \
+                            ((hl.is_defined(mt4['allele_type'])) & (mt4['allele_type']=='ins') & \
+                             (mt4['AS_VQSLOD'] < mt4['filtering_model'].indel_cutoff.min_score)) | \
+                            ((hl.len(mt4.alleles) < 2) | (hl.len(mt4.alleles) > 2)) | \
+                            ((hl.len(mt4.alleles)==2) & (mt4.n_unsplit_alleles != 2)) | \
+                            (hl.is_missing(mt4['InbreedingCoeff'])) | \
+                            ((hl.is_defined(mt4['InbreedingCoeff'])) & (mt4['InbreedingCoeff'] < -0.3))
+        mt5 = mt4.filter_rows(filter_conditions, keep=False)
+         
+        # Restricted to high quality variants (GQ>=20, DP>=10)
+        filter_condition = (mt5.DP >= 10) & (mt5.GQ >= 20)
+        mt6 = hl.variant_qc(mt5.filter_entries(filter_condition, keep=True))
+
+        # Section 4 - deCODE specific filter
+       
+        # Exclude variants with call rate < 0.99 (not in deCODE paper)
+        mt7 = mt6.filter_rows(mt6.variant_qc.call_rate >= 0.99)
+
+        # Identify singleton mutations without filters 
+        mt8 = mt7.filter_rows(mt7.variant_qc.n_non_ref == 1)
+
+        # Identify singleton mutations with filters
+          # sequence depth >= 16 & GQ >= 90
+          # >=3 indepedent reads containing a variant allele required
+        filter_condition = (mt8.DP >= 16) & (mt8.GQ >= 90) & (mt8.AD[1] >= 3)
+        mt9 = hl.variant_qc(mt8.filter_entries(filter_condition, keep=True))
+        mt10 = mt9.filter_rows(mt9.variant_qc.n_non_ref == 1) 
+        
+        # gnomAD allele frequency 
+        ref_ht = ('gs://cpg-reference/seqr/v0-1/combined_reference_data_grch38-2.0.4.ht')
+        ref_ht = hl.read_table(ref_ht)
+      
+        # Annotate variants with CADD scores, gnomAD etc.
+        mt11 = mt10.annotate_rows(
+                    cadd=ref_ht[mt10.row_key].cadd,
+                    gnomad_genomes=ref_ht[mt10.row_key].gnomad_genomes,
+                    gnomad_genome_coverage=ref_ht[mt10.row_key].gnomad_genome_coverage
+               )
+        
+        del ref_ht
+
+        # Apply gnomAD AF filter
+        AF_cutoff = 1/(11262*2)
```

Comment:
Could the sample number(?) of 11262 be pulled out from the input data somewhere rather than hard-coding it?

Reply:
Thanks for the suggestion! Done!